### PR TITLE
Search in A 2D row wise sorted matrix in O(log(N) time  using Binary Search

### PR DESCRIPTION
**Pull request by PRA-developer-TIK**
This is a most efficient problem reducing time complexity to O(log(N)) from normal O(N^2) and well commented explaination

### CODE REVIEW

**Comment by PRA-developer-TIK** [comment]
Will This be considered for hacktoberfest ??

File: C++-Programs/Search2DMatrix.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            end = mid - 1;
+        else
+            start = mid + 1;
+    }
+    return 0;
+}
+
+int main()
+{
+    int m, n, B, val;
```
Comment (PrajwalCH) [inline C++-Programs/Search2DMatrix.cpp:52]: Use good variable names. It's quite hard to figure out what **m, n, B and val***  are for.

**Review by PrajwalCH — COMMENTED** [review]
Use the proper name for variable. Don't use single letter.

**Comment by PRA-developer-TIK** [comment]
Yepp will do the required changes and then make a PR

**Review by PrajwalCH — APPROVED** [review]

**Pull request merged**